store/postgres/src/chain_store.rs

```diff
@@ -1813,6 +1841,164 @@ impl ChainStoreTrait for ChainStore {
     }
 }
 
+mod recent_blocks_cache {
+    use super::*;
+    use std::collections::BTreeMap;
+
+    struct CachedBlock {
+        ptr: BlockPtr,
+        data: Option<json::Value>,
+        parent_hash: BlockHash,
+    }
+
+    struct Inner {
+        // Note: At all times, only a continuous range of blocks will be in the
+        // cache, i.e. without gaps. This constraint makes it possible to easily
+        // verify a direct line of ancestry between two blocks via parent hashes.
+        blocks: BTreeMap<BlockNumber, CachedBlock>,
+        // We only store these many blocks.
+        capacity: usize,
+    }
+
+    impl Inner {
+        fn get_block(
+            &self,
+            child: &BlockPtr,
+            offset: BlockNumber,
+        ) -> Option<(&BlockPtr, Option<&json::Value>)> {
+            // Before we can go find the ancestor, we need to make sure that
+            // we're looking for the ancestor of the right block, i.e. check if
+            // the hash (and number) of the child matches.
+            let child_is_cached = &self.blocks.get(&child.number)?.ptr == child;
+
+            if child_is_cached {
+                let ancestor_block_number = child.number - offset;
+                let block = self.blocks.get(&ancestor_block_number)?;
+                Some((&block.ptr, block.data.as_ref()))
+            } else {
+                None
+            }
+        }
+
+        fn chain_head(&self) -> Option<&BlockPtr> {
+            self.blocks.iter().next_back().map(|b| &b.1.ptr)
+        }
+
+        fn last_block(&self) -> Option<&CachedBlock> {
```

Comment:
I got a little confused here - isn't that the first block, i.e., the one with the lowest block number? Calling this `first_block` or `earliest_block` would be clearer